### PR TITLE
Initial Commit

### PR DESCRIPTION
**Pull request by adamhf**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+		}
+
+		fmt.Printf("%s: %+v\n", createdIssue.Key, createdIssue.Self)
+
+		issuesCreatedCounter.Inc()
+		log.Printf("Created Jira Issue '%s' in project: %s' \n", createdIssue.Key, feed.JiraProjectID)
+	}
+}
+
+func readConfig(path string) *Config {
```
Comment (nabadger) [inline main.go:169]: I probably would have looked at https://github.com/spf13/viper for config/env-var handling.

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      - /app/wait-for-it.sh
+      - ${REDIS_URL}
+      - --timeout=60
+      - --strict
+      - --
+      - /app/rss_sync
+    restart: unless-stopped
+    ports:
+      - "8080:8080"
+  db:
```
Comment (nabadger) [inline docker-compose.yaml:21]: maybe this should be `redis` instead of `db` ?

File: config.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+interval: 300
```
Comment (nabadger) [inline config.yaml:1]: Not sure if this is meant to be committed?

File: main.go [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+func initialise(env EnvValues) (redisClient *redis.Client, jiraClient *jira.Client, config *Config) {
+	gaugeOpts := prometheus.GaugeOpts{
+		Name: "last_run_time",
+		Help: "Last Run Time in Unix Seconds",
+	}
+	lastRunGauge = prometheus.NewGauge(gaugeOpts)
+	prometheus.MustRegister(lastRunGauge)
+
+	issuesCreatedCounterOpts := prometheus.CounterOpts{
+		Name: "issue_creation_total",
```
Comment (nabadger) [inline main.go:194]: We probably want to look at using options to specify namespace/sub-system if applicable:

examples:

- https://github.com/prometheus/client_golang/blob/master/prometheus/examples_test.go#L35
- https://github.com/prometheus/client_golang/blob/master/prometheus/examples_test.go#L55

File: config.yaml (hunk 3 above)
Comment (adamhf) [reply config.yaml:1]: I committed it as I use it whilst testing and would use the same file again in the future, it contains no private information so figured why not.

File: main.go (hunk 1 above)
Comment (adamhf) [reply main.go:169]: Looks neat, I copied most of this from the GitlabRSSSync and modified it to suit our purposes.  If I do another rework I'll definitely consider this.

File: main.go (hunk 4 above)
Comment (adamhf) [reply main.go:194]: Cool, I'll add the Subsystem

File: main.go [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"jaytaylor.com/html2text"
+	"log"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+	"time"
+)
```
Comment (jtdoepke) [inline main.go:20]: Style nitpick: there's a convention for imports to have three sections (or more if the situation calls for it):

```
import (
    STDLIB_PACKAGES

    THIRD_PARTY_PACKAGES

    PACKAGES_FROM_ELSEWHERE_IN_THIS_REPO
)
```

File: main.go [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+
+var addr = flag.String("listen-address", ":8080", "The address to listen on for HTTP requests.")
+var lastRunGauge prometheus.Gauge
+var issuesCreatedCounter prometheus.Counter
+var issueCreationErrorCounter prometheus.Counter
```
Comment (jtdoepke) [inline main.go:25]: Style nitpick: these could be grouped into a single var declaration.
```suggestion
var (
    lastRunGauge prometheus.Gauge
    issuesCreatedCounter prometheus.Counter
    issueCreationErrorCounter prometheus.Counter
)
```

File: main.go [hunk 7]
```diff
@@ -0,0 +1,334 @@
...
+	} else if len(issues) > 1 {
+		retVal = true
+		log.Printf("Found multiple issues that match \"%s\":", itemTitle)
+		for _, x := range issues {
+			log.Printf("%s ", x.Key)
+		}
+	} else {
+		retVal = true
+	}
+	return retVal
```
Comment (jtdoepke) [inline main.go:76]:
```suggestion
	if len(issues) == 0 {
		return false
	} else if len(issues) > 1 {
		log.Printf("Found multiple issues that match \"%s\":", itemTitle)
		for _, x := range issues {
			log.Printf("%s ", x.Key)
		}
	}
	return true
```

File: main.go [hunk 8]
```diff
@@ -0,0 +1,334 @@
...
+}
+
+func initialise(env EnvValues) (redisClient *redis.Client, jiraClient *jira.Client, config *Config) {
+	gaugeOpts := prometheus.GaugeOpts{
+		Name: "last_run_time",
+		Subsystem: "jira_rss_sync",
+		Help: "Last Run Time in Unix Seconds",
+	}
+	lastRunGauge = prometheus.NewGauge(gaugeOpts)
+	prometheus.MustRegister(lastRunGauge)
```
Comment (jtdoepke) [inline main.go:192]: You could save a few lines here with the [promauto](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/promauto) package.

File: main.go [hunk 9]
```diff
@@ -0,0 +1,334 @@
...
+		redisClient = redis.NewFailoverClient(&redis.FailoverOptions{
+			SentinelAddrs: []string{env.RedisURL},
+			Password:      env.RedisPassword,
+			MasterName:    "mymaster",
+			DB:            0, // use default DB
+		})
+	}
+
+	if err := redisClient.Ping().Err(); err != nil {
+		panic(fmt.Sprintf("Unable to connect to Redis @ %s", env.RedisURL))
```
Comment (jtdoepke) [inline main.go:239]: You could do a `log.Panicf()` here.

File: main.go [hunk 10]
```diff
@@ -0,0 +1,334 @@
...
+			for _, configEntry := range config.Feeds {
+				configEntry.checkFeed(redisClient, jiraClient)
+			}
+			lastRunGauge.SetToCurrentTime()
+			time.Sleep(time.Duration(config.Interval) * time.Second)
+		}
+	}()
+
+	http.Handle("/metrics", promhttp.Handler())
+	log.Fatal(http.ListenAndServe(*addr, nil))
```
Comment (jtdoepke) [inline main.go:263]: It seems like this and the healthcheck server could be combined.

**Pull request merged**